core/tests/test_utils.py

```diff
@@ -397,16 +446,35 @@ def set_collection_editors(self, collection_editor_usernames):
             self.set_user_role(name, feconf.ROLE_ID_COLLECTION_EDITOR)
 
     def get_current_logged_in_user_id(self):
+        """Gets the user_id of the current logged-in user.
+
+        Returns:
+            str. Value of USER_ID env variable.
+        """
         return os.environ['USER_ID']
 
     def get_user_id_from_email(self, email):
+        """Gets the user_id corresponding to the given email.
+
+        Args:
+            email: str. valid email id stored in the App Engine database.
+
+        Returns:
+            user_id: str. Id of user possessing the given email.
+        """
         return current_user_services.get_user_id_from_email(email)
 
     def save_new_default_exploration(
             self, exploration_id, owner_id, title='A title'):
         """Saves a new default exploration written by owner_id.
 
-        Returns the exploration domain object.
+        Args:
+            exploration_id: str. The id of the new validated exploration.
+            owner_id: str. The user_id of the creator of the exploration.
+            title: str. The title of the exploration.
+
+        Returns:
+            Collection. The exploration domain object.
```

Comment:
The type of an exploration domain object is "Collection"?

Reply:
Done!

Reply:
You are writing docstrings that other devs will refer to in the future, so don't guess. Try running one of the backend tests that references this, and do something like `raise Exception(type(return_value))` here to figure out the type.

Reply:
I am not finding a way to find it out. Which file should I run tests on? Please suggest some ways.

Reply:
Search the codebase. What calls this method? Once you've figured that out, search the codebase again: what calls that method? Continue until you find a test that you can run using the back end test suite. The wiki has instructions for how to run a single test.

Reply:
I have tried printing its type but its not getting printed. Only tests are shown either passed or failed. I have even tried executing python file `learner_dashboard_tests.py` but its giving errors. Please help!

Reply:
Earlier, I suggested doing `raise Exception(...)` -- see a previous comment in this thread. This should surface the type in the test logs.

Reply:
Its type is- Collection as evident from the error logs. 
![screen shot 2018-01-01 at 5 57 22 pm](https://user-images.githubusercontent.com/26625980/34467637-854713ac-ef1d-11e7-8705-1508dadb5c94.png)

Reply:
Thanks for checking. I'm a little confused because I can't see where `collection` is defined in this method. Where exactly did you add the `raise Exception` line to? It should wrap the returned value.

Reply:
Here is the attached image of the function where I have added this. Actually this function is calling another function  `collection_domain.Collection.create_default_collection` and this function is returning the collection.
![screen shot 2018-01-02 at 11 04 23 am](https://user-images.githubusercontent.com/26625980/34475014-37f76210-efad-11e7-82ff-cbb69a609e09.png)

Reply:
Are we looking at the same function? I am talking about this one, where this comment is located:

https://github.com/oppia/oppia/pull/4280/files#diff-6f3c73d302248e02f5b3eeb202eea43dR470

It looks different from your screenshot.

Reply:
Yes, we are looking at the same function `save_new_default_collection`. I ran tests on `learner_dashboard_test.py` which is using this function. I added `raise Exception(...)` as in the above screenshot and hence this exception is raised. Please tell me if I am unable to answer your question.
![screen shot 2018-01-01 at 5 57 22 pm](https://user-images.githubusercontent.com/26625980/34477098-fac1bd18-efbc-11e7-9b4f-096c5e2eb84a.png)

Reply:
OK, for some reason we are looking at different things, and I'm not sure I understand why.

On line 470 on this page (after your changes were made), where the present comment thread is located:

    https://github.com/oppia/oppia/pull/4280/files#diff-6f3c73d302248e02f5b3eeb202eea43dR470

I am reading the function as save_new_default_exploration(). The save_new_default_collection() function is on line 602, and I have not left any comments there. I have also left a comment on save_new_valid_exploration(). Do you see something different in the GitHub "Files Changed" tab?

Reply:
Thanks @seanlip for pointing it out. I was looking at other function. Yes, its type should be Exploration in both the functions. I have seen in the test logs. Thanks!
![screen shot 2018-01-02 at 2 41 51 pm](https://user-images.githubusercontent.com/26625980/34479185-2c9155de-efcb-11e7-9aae-b4cf47671560.png)

Reply:
Excellent; apart from that, everything else you were doing seems to be along the right lines.

Do you know why you were looking at the other function? I thought I'd pointed to this one quite clearly but it'd be good to know if GitHub's misbehaving.

Reply:
There seems to be difference in lines before this updation and after that. Actually I was referring  to the previous updates and thus, ended up with different function. Moreover, the docstrings for both the functions are more or less same and that confirmed me that I was referring to the same function.